wagtail/images/models.py

```diff
@@ -422,24 +443,91 @@ def get_rendition(self, filter: Union["Filter", str]) -> "AbstractRendition":
         except Rendition.DoesNotExist:
             rendition = self.create_rendition(filter)
             # Reuse this rendition if requested again from this object
-            if "renditions" in getattr(self, "_prefetched_objects_cache", {}):
-                self._prefetched_objects_cache["renditions"]._result_cache.append(
-                    rendition
-                )
-            elif hasattr(self, "prefetched_renditions"):
-                self.prefetched_renditions.append(rendition)
-
+            self._add_to_prefetched_renditions(rendition)
         try:
             cache = caches["renditions"]
-            key = Rendition.construct_cache_key(
-                self.id, filter.get_cache_key(self), filter.spec
-            )
-            cache.set(key, rendition)
         except InvalidCacheBackendError:
             pass
+        else:
+            # Avoid additional query for key generation
+            rendition.image = self
+            cache.set(rendition.cache_key, rendition)
 
         return rendition
 
+    def get_renditions(self, *filter_specs: str) -> Dict[str, "AbstractRendition"]:
```

Comment:
Question: Is it possible to reuse this implementation for `get_rendition` (singular) too? The implementations are very similar, and it would remove a lot of nearly-duplicate code.

Reply:
Nice suggestion there @RealOrangeOne! Implemented in 18a7b4d